File: Python/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]


print(Solution.twoSum(Solution, [2, 7, 11, 15], 9)) # [0, 1]

'''
Runtime: 8025 ms, faster than 5.07% of Python online submissions for Two Sum.
Memory Usage: 14.3 MB, less than 67.65% of Python online submissions for Two Sum.
'''

File: C/TwoSum.c
#include <stdio.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

int *twoSum(int *nums, int numsSize, int target, int *returnSize) {
    *returnSize = 2;
    int i = 0, j = 0;
    int *result = (int *) malloc(sizeof(int) * 2);
    for (i = 0; i < numsSize; i++) {
        for (j = i + 1; j < numsSize; j++) {
            if (target == nums[i] + nums[j]) {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }
    return result;
}

int main() {
    int nums[5] = {1, 9, 2, 8, 2};
    int returnSize = 2;

    int *ret = twoSum(nums, 5, 17, &returnSize);
    for (int i = 0; i < returnSize; i++) {
        printf("%d ", ret[i]);
    }
}

/**
 * Runtime: 131 ms, faster than 77.46% of C online submissions for Two Sum.
 * Memory Usage: 6.5 MB, less than 28.10% of C online submissions for Two Sum.
 */
